components/tidb_query_expr/src/impl_string.rs

```diff
@@ -499,6 +499,22 @@ pub fn upper(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
     Ok(writer.write_ref(Some(arg)))
 }
 
+#[rpn_fn(writer)]
+#[inline]
+pub fn lower_utf8(arg: BytesRef, writer: BytesWriter) -> Result<BytesGuard> {
+    match str::from_utf8(arg) {
+        Ok(s) => Ok(writer.write_ref(Some(s.to_lowercase().as_bytes()))),
+        Err(err) => Err(box_err!("invalid input value: {:?}", err)),
```

Comment:
As https://github.com/tikv/tikv/pull/9640 has been merged, please use `str::from_utf8(arg)?` to return error.

Reply:
Done :+1: